app/src/pages/CardSubmission/WaitingForSubmissions.tsx

```diff
@@ -37,77 +37,78 @@ function WaitingForSubmissions() {
     return null
   }
 
-  const allPlayersSubmitted = total !== 0 && submittedSoFar === total
+  let state: WaitingForSubmissionsState
+
+  if (total === 0 || submittedSoFar !== total) {
+    state = WaitingForSubmissionsState.Waiting
+  } else if (PlayerRole.Host === currentPlayer.role) {
+    state = WaitingForSubmissionsState.SubmittedAssign
+  } else {
+    state = WaitingForSubmissionsState.SubmittedWait
+  }
+
+  const unscreenedCards = currentGame.cards.filter(function (card) {
+    return (
+      PlayerRole.Host === currentPlayer.role &&
+      currentGame.screen_cards &&
+      null === card.is_allowed
+    )
```

Comment:
We'll probably want to exclude the host's cards from this set?

Reply:
yep - i think that's better 👍, nbd either way